src/api/Main/BuisnessLayer/Shrooms.Domain/Services/UserService/UserService.cs

```diff
@@ -150,10 +153,30 @@ public void Delete(string userToDelete, UserAndOrganizationDTO userOrg)
             UnassignUserFromWalls(userToDelete, userOrg.OrganizationId);
             _userManager.RemoveLogins(userToDelete);
 
+            await Anonymize(user, userOrg);
+
             _usersDbSet.Remove(user);
             _uow.SaveChanges(userOrg.UserId);
         }
 
+        private async Task Anonymize(ApplicationUser user, UserAndOrganizationDTO userOrg)
+        {
+            await _pictureService.RemoveImage(user.PictureId, userOrg.OrganizationId);
+
+            user.Email = string.Empty;
+            user.FirstName = string.Empty;
+            user.LastName = string.Empty;
+            user.Bio = string.Empty;
+            user.PhoneNumber = string.Empty;
+            user.UserName = string.Empty;
+            user.FacebookEmail = string.Empty;
+            user.GoogleEmail = string.Empty;
+            user.PictureId = string.Empty;
+            user.BirthDay = DateTime.UtcNow;
+
+            _uow.SaveChanges(userOrg.UserId);
```

Comment:
Is that SaveChanges really needed here, if yes - then why?
"Final" SaveChanges are done in upper Delete method - isn't it enough?

Reply:
Yes, it is needed, because when soft deleting entity's state is set to _detached_ which means all unsaved changes to an entity are lost.